### ISSUE TITLE
Feature: Possibility to generate a manifest/map/lookup

### ISSUE THREAD

**Issue by pellebjerkestrand** [asbjornh/view-models#5]
I'll use C#/Razor as an example here.

After generating the classes, one still has to stringly type the path to the component in the global scope:

```
@Html.React("Components.Header", model.Header)
```

It would be nice to be able to generate something to remove this, where the something generator would receive a list of the metadata for all processed components.

We could then generate a static class

```
public static class ComponentPaths
{
   public static string Header => "Components.Header";
}
```

and do:

```
@Html.React(ComponentPaths.Header, model.Header)
```

This will give us compile time safety for the now stringly typed component paths.

**Comment by asbjornh** [comment]
I could add a callback function option to the webpack plugin that exposes the metadata. Would that cover the use case?

**Comment by pellebjerkestrand** [comment]
It would give developers a hook to implement this functionality themselves, at least.

**Comment by asbjornh** [comment]
Alternatively `view-models` could provide a manifest generator, but it would then have to look like this:

```cs
public static class ViewModelManifest
{
  public static string Header = "Header";
}
```

You could then probably do something like this:

```cs
@Html.React("Components." + ViewModelManifest.Header, model.Header)
```

**Comment by pellebjerkestrand** [comment]
What about exposing a callback, and using the result of that callback to write a file?

The expected shape of the return value could be something like:

```js
{
  path: '',
  data: ''
}
```

The callback would then not have to implement anything file system related, while being able to specify the content and location of the file exactly.

**Comment by asbjornh** [comment]
Sure! Good idea. I'll have a look later

**Comment by asbjornh** [comment]
Fixed in `1.1.0` by e9a831d8dcb6432b4387536e594dc20be92f333a

`metaFileGenerator` is documented [here](https://github.com/asbjornh/view-models/blob/master/docs/webpack-plugin.md)
